nixos/modules/services/networking/iperf.nix

```diff
@@ -0,0 +1,92 @@
+{ config, lib, pkgs, ... }: with lib;
+let
+  cfg = config.services.iperf3;
+
+  api = {
+    enable = mkEnableOption "iperf3 network throughput testing server";
+    port = mkOption {
+      type        = types.int;
+      default     = 5201;
+      description = "Server port to listen on for iperf3 client requsts";
+    };
+    affinity = mkOption {
+      type        = types.nullOr types.string;
+      default     = null;
+      description = "CPU affinity for the process";
+    };
+    bind = mkOption {
+      type        = types.nullOr types.string;
+      default     = null;
+      description = "Bind to the specific interface associated with the given address";
+    };
+    verbose = mkOption {
+      type        = types.bool;
+      default     = false;
+      description = "Give more detailed output";
+    };
+    forceFlush = mkOption {
+      type        = types.bool;
+      default     = false;
+      description = "Force flushing output at every interval";
+    };
+    debug = mkOption {
+      type        = types.bool;
+      default     = false;
+      description = "Emit debugging output";
+    };
+    rsaPrivateKey = mkOption {
+      type        = types.nullOr types.string;
+      default     = null;
+      description = "Path to the RSA private key (not password-protected) used to decrypt authentication credentials from the client";
+    };
+    authorizedUsersFile = mkOption {
+      type        = types.nullOr types.string;
+      default     = null;
+      description = "Path to the configuration file containing authorized users credentials to run iperf tests";
+    };
+    extraFlags = mkOption {
+      type        = types.listOf types.string;
+      default     = [ ];
+      description = "Extra flags to pass to iperf3(1)";
+    };
+  };
+
+  imp = {
+    systemd.services.iperf3 = {
+      description = "iperf3 daemon";
+      unitConfig.Documentation = "man:iperf3(1) https://iperf.fr/iperf-doc.php";
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network.target" ];
+
+      serviceConfig = {
+        Type = "simple";
+        Restart = "on-failure";
+        RestartSec = 2;
+        DynamicUser = true;
+        RuntimeDirectory = "iperf3";
+        PIDFile = "/run/iperf3/iperf.pid";
+        PrivateDevices = true;
+        CapabilityBoundingSet = "";
+        NoNewPrivileges = true;
+      };
+
+      script = ''
+        exec ${pkgs.iperf3}/bin/iperf \
+          --server \
+          --pidfile /run/iperf3/iperf.pid \
```

Comment:
It is preferred to not use PID files if not necessary, which is probably the case here, because iperf runs in the foreground by default, which is good.